### PR TITLE
Show all configurations in "abstract" printer from cloud

### PR DESCRIPTION
**Pull request by nallath**
CURA-8463

### CODE REVIEW

File: plugins/UM3NetworkPrinting/src/Cloud/CloudApiClient.py [hunk 1]
```diff
@@ -77,6 +76,24 @@ def getClusters(self, on_finished: Callable[[List[CloudClusterResponse]], Any],
...
+    def getClustersByMachineType(self, machine_type, on_finished: Callable[[List[CloudClusterResponse]], Any], failed: Callable) -> None:
+        # HACK: There is something weird going on with the API, as it reports printer types in formats like
+        # "ultimaker_s3", but wants "Ultimaker S3" when using the machine_variant filter query. So we need to do some
+        # conversion!
+
+        machine_type = machine_type.replace("_plus", "+")
+        machine_type = machine_type.replace("_", " ")
+        machine_type = machine_type.replace("ultimaker", "ultimaker ")
+        machine_type = machine_type.replace("  ", " ")
+        machine_type = machine_type.title()
```
Comment (casperlamboo) [inline plugins/UM3NetworkPrinting/src/Cloud/CloudApiClient.py:90]: why not use the prettified machine name here?

File: plugins/UM3NetworkPrinting/src/Cloud/CloudOutputDeviceManager.py [hunk 2]
```diff
@@ -373,22 +380,33 @@ def _connectToActiveMachine(self) -> None:
...
-            if device.key == stored_cluster_id:
-                # Connect to it if the stored ID matches.
-                self._connectToOutputDevice(device, active_machine)
-            elif local_network_key and device.matchesNetworkKey(local_network_key):
-                # Connect to it if we can match the local network key that was already present.
-                self._connectToOutputDevice(device, active_machine)
-            elif device.key in output_device_manager.getOutputDeviceIds():
-                # Remove device if it is not meant for the active machine.
-                output_device_manager.removeOutputDevice(device.key)
+        if active_machine.getMetaDataEntry("is_abstract_machine") != "True":
```
Comment (casperlamboo) [inline plugins/UM3NetworkPrinting/src/Cloud/CloudOutputDeviceManager.py:386]:
```suggestion
        if not parseBool(active_machine.getMetaDataEntry("is_abstract_machine")):
```

File: plugins/UM3NetworkPrinting/src/Cloud/AbstractCloudOutputDevice.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    def __init__(self, api_client: CloudApiClient, printer_type: str, parent: QObject = None) -> None:
+
+        self._api = api_client
+        properties = {
+            #b"address": cluster.host_internal_ip.encode() if cluster.host_internal_ip else b"",
+           # b"name": cluster.friendly_name.encode() if cluster.friendly_name else b"",
+            ##b"firmware_version": cluster.host_version.encode() if cluster.host_version else b"",
+            b"printer_type": printer_type.encode()
+            #b"cluster_size": str(cluster.printer_count).encode() if cluster.printer_count else b"1"
```
[the remark this answers is not in the document]
Comment (nallath) [reply plugins/UM3NetworkPrinting/src/Cloud/AbstractCloudOutputDevice.py:30]: Removed! It's a bit of legacy on my side

File: plugins/UM3NetworkPrinting/src/Cloud/CloudApiClient.py [hunk 4]
```diff
@@ -71,21 +71,39 @@ def getClusters(self, on_finished: Callable[[List[CloudClusterResponse]], Any],
...
-        url = "{}/clusters?status=active".format(self.CLUSTER_API_ROOT)
+        url = f"{self.CLUSTER_API_ROOT}/clusters?status=active"
         self._http.get(url,
                        scope = self._scope,
                        callback = self._parseCallback(on_finished, CloudClusterResponse, failed),
                        error_callback = failed,
                        timeout = self.DEFAULT_REQUEST_TIMEOUT)
 
+    def getClustersByMachineType(self, machine_type, on_finished: Callable[[List[CloudClusterWithConfigResponse]], Any], failed: Callable) -> None:
+        # HACK: There is something weird going on with the API, as it reports printer types in formats like
```
Comment (rburema) [inline plugins/UM3NetworkPrinting/src/Cloud/CloudApiClient.py:82]: There's a `product_to_id.json` in the XmlMaterialProfile plugin ;-p

Comment (nallath) [reply plugins/UM3NetworkPrinting/src/Cloud/CloudApiClient.py:82]: Yeah, the downside of that is that it doesn't work for all machines (so only for machines that are explicitly in that list)

**Pull request merged**